src/averageAges.js

```diff
@@ -55,6 +81,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
+  let peopleArr = [];
+
+  if (onlyWithSon) {
+    peopleArr = people.filter(person => person.sex === 'm');
+  } else {
+    peopleArr = people;
+  }
+
+  const mothers = people.filter(mother =>
+    mother.sex === 'f' && peopleArr.some(person =>
+      person.mother === mother.name));
+
+  const ageDifferences = mothers.flatMap(mother => {
+    const children = peopleArr.filter(child => child.mother === mother.name);
+
+    return children.map(child => child.born - mother.born);
+  });
```

Comment:
This is correct, but i think it would be far more readable to get `childrenSum`, `mothersSum`, calculate the difference between them and divide by number of people.

``` 
x, y ->  children ages, 
a, b -> mothers ages

then: (a - x + b - y)  / 2 === (a + b - (x + y)) / 2
```

Reply:
Try my solution and then decide which one do you like more.